src/gui/processing/qgsprocessingoutputdestinationwidget.cpp

```diff
@@ -728,4 +765,25 @@ void QgsProcessingLayerOutputDestinationWidget::dropEvent( QDropEvent *event )
   leText->setHighlighted( false );
 }
 
+bool QgsProcessingLayerOutputDestinationWidget::isTemporaryLayer( const QString &value )
+{
+  if ( value == QgsProcessing::TEMPORARY_OUTPUT || value.isEmpty() || mParameter->type() != QgsProcessingParameterFeatureSink::typeName() || !mParameter->supportsNonFileBasedOutput() )
+    return false;
+
+  QString provider;
+  QString uri;
+  bool hasProviderAndUri = QgsProcessingUtils::decodeProviderKeyAndUri( value, provider, uri );
+
+  if ( provider.startsWith( QStringLiteral( "memory" ) ) )
```

Comment:
A plugin provider could potentially start with `memory` so better be explicit
```suggestion
  if ( provider == QLatin1String( "memory" ) )
```